submissions/2024-10-11-MLPepper-RECAP-Optimized-Fragments-Add-Iodines-v1.0/create_singlepoints.py

```diff
@@ -0,0 +1,45 @@
+from openff.qcsubmit.results import OptimizationResultCollection
+from openff.qcsubmit.results.filters import ConnectivityFilter, RecordStatusFilter, RecordStatusEnum
+from openff.qcsubmit.common_structures import QCSpec, SCFProperties
+# from openff.qcsubmit.common_structures import DDXSettings, SCFProperties, Metadata
+from qcportal import PortalClient
+
+DATASET_NAME  = ""
+ADDRESS = ""
+USERNAME = ""
+PASSWORD = ""
+client = PortalClient(address=ADDRESS, username=USERNAME, password=PASSWORD)
+
+result_dataset = OptimizationResultCollection.from_server(client=client, datasets=DATASET_NAME, spec_name="aimnet2-wb97m-d3")
+print(result_dataset)
+print(f"Dataset has {result_dataset.n_results} raw results.")
+
+status_filter = RecordStatusFilter(status=RecordStatusEnum.complete)
+result_dataset = result_dataset.filter(status_filter)
+
+print(f"Dataset has {result_dataset.n_results} complete records.")
+
+# remove connectivity issues
+connect_filter = ConnectivityFilter()
+result_dataset = result_dataset.filter(connect_filter)
+
+print(f"Dataset has {result_dataset.n_results} after removing connection issues")
+# mock some data for the dataset will be removed later
+
+print(f"Building single point dataset")
+basic_dataset = result_dataset.create_basic_dataset(
```

Comment:
Just an FYI for next time--you could have just created the dataset here, by giving it the name/metadata here and adding the QCSpecs to this dataset. It seems to work fine the way you did it as well, but just wanted to give you a heads up.

Reply:
Cool thanks for the heads up!